k8s/supportability/src/collect/logs/mod.rs

```diff
@@ -109,95 +113,114 @@ impl LogCollection {
-                        .get(KUBERNETES_HOST_LABEL_KEY)
-                        .ok_or_else(|| {
-                            K8sResourceError::invalid_k8s_resource_value(format!(
-                                "Hostname not found for node {}",
-                                node_name.clone()
-                            ))
-                        })?
-                        .clone(),
-                );
-            }
-            // Since pod object fetched from Kube-apiserver there will be always
-            // spec associated to pod
-            let containers = pod
-                .spec
-                .ok_or_else(|| {
-                    K8sResourceError::invalid_k8s_resource_value("Pod sepc not found".to_string())
-                })?
-                .containers;
-
-            for container in containers {
-                logging_resources.insert(LogResource {
-                    container_name: container.name,
-                    host_name: hostname.clone(),
-                    label_selector: format!("app={}", service_name.clone()),
-                    service_type: service_name.clone(),
-                });
+                    .get(KUBERNETES_HOST_LABEL_KEY)
+                    .ok_or_else(|| {
+                        K8sResourceError::invalid_k8s_resource_value(format!(
+                            "Hostname not found for node {}",
+                            node_name.clone()
+                        ))
+                    })?
+                    .clone(),
+            );
+        }
+        // Since pod object fetched from Kube-apiserver there will be always
+        // spec associated to pod
+        let containers = pod
+            .spec
+            .ok_or_else(|| {
+                K8sResourceError::invalid_k8s_resource_value("Pod spec not found".to_string())
+            })?
+            .containers;
+
+        for container in containers {
+            logging_resources.insert(LogResource {
+                container_name: container.name,
+                host_name: hostname.clone(),
+                label_selector: format!("app={}", service_name.clone()),
+                service_type: service_name.clone(),
+            });
+        }
+        Ok(logging_resources)
+    }
+
+    async fn get_logging_resources(
+        &self,
+        pods: Vec<Pod>,
+    ) -> Result<HashSet<LogResource>, LogError> {
+        let nodes_map = self
+            .k8s_logger_client
+            .get_k8s_clientset()
+            .get_nodes_map()
+            .await?;
+        let mut logging_resources = HashSet::new();
+
+        for pod in pods {
+            match self
+                .pod_logging_resources(pod.clone(), nodes_map.clone())
+                .await
+            {
+                Ok(resources) => logging_resources.extend(resources),
+                Err(error) => log(format!(
+                    "Skipping the pod in pending state: {:?} error : {:?}",
+                    pod.metadata.name, error
```

Comment:
Not just pending, can be any error

```suggestion
                    "Skipping the pod {:?} due to error: {error:?}", pod.metadata.name
```